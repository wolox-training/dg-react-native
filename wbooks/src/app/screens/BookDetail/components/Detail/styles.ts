import { StyleSheet } from 'react-native';
import { COLORS } from '@constants/colors';

export default StyleSheet.create({
  container: {
    backgroundColor: COLORS.background,
    flex: 1,
    paddingVertical: 10
  },
  subContainer: {
    flex: 1,
    padding: 20
  },
  detailsContainer: {
    paddingHorizontal: 30,
    paddingVertical: 15,
    backgroundColor: COLORS.white,
    borderRadius: 5,
    shadowColor: COLORS.black,
    shadowOffset: {
      width: 0,
      height: 1
    },
    shadowOpacity: 0.18,
    shadowRadius: 1,
    elevation: 1
  },
  details: {
    flexDirection: 'row',
    marginBottom: 30
  },
  image: {
    width: 69,
    height: 105
  },
  title: {
    fontSize: 24,
    color: COLORS.light,
    paddingRight: 50,
    paddingLeft: 20,
    marginBottom: 5
  },
  generalText: {
    paddingLeft: 20,
    fontSize: 15,
    color: COLORS.light
  }
});
