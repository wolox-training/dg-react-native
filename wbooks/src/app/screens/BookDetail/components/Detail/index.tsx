import React from 'react';
import { Image, SafeAreaView, ScrollView, View, Text } from 'react-native';
import { Book } from '@interfaces/Book';
import { validateImageUrl } from '@utils/bookUtils';
import Button from '@components/Button';

import styles from './styles';

interface Props {
  book: Book;
}

function Detail({ book }: Props) {
  const { imageUrl, title, author, year, genre } = book;
  const source = validateImageUrl(imageUrl);

  return (
    <SafeAreaView style={styles.container}>
      <ScrollView>
        <View style={styles.subContainer}>
          <View style={styles.detailsContainer}>
            <View style={styles.details}>
              <Image
                accessibilityLabel="detailImage"
                resizeMode="stretch"
                source={source}
                style={styles.image}
              />
              <View>
                <Text numberOfLines={1} style={styles.title}>
                  {title}
                </Text>
                <Text style={styles.generalText}>{author}</Text>
                <Text style={styles.generalText}>{year}</Text>
                <Text style={styles.generalText}>{genre}</Text>
              </View>
            </View>
            <Button title="ADD TO WISHLIST" />
            <Button title="RENT" type="secondary" style={styles.rentButton} />
          </View>
        </View>
      </ScrollView>
    </SafeAreaView>
  );
}

export default Detail;
