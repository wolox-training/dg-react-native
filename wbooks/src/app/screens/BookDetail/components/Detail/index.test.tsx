import React from 'react';
import { render } from '@testing-library/react-native';
import { BOOKS_MOCK } from '@constants/mockBooks';
import bookPlaceholder from '@assets/book/img_book_placeholder.png';

import Detail from './index';

describe('Detail test suit', () => {
  it('Render correctly', () => {
    const book = render(<Detail book={BOOKS_MOCK[0]} />);
    expect(book).toMatchSnapshot();
  });
  it('Render with all properties and no image uri', () => {
    const { getByLabelText, getByText } = render(<Detail book={BOOKS_MOCK[0]} />);
    const image = getByLabelText('detailImage');
    expect(image.props.source).toEqual(bookPlaceholder);
    expect(getByText(BOOKS_MOCK[0].title)).toBeTruthy();
    expect(getByText(BOOKS_MOCK[0].author)).toBeTruthy();
    expect(getByText(BOOKS_MOCK[0].year)).toBeTruthy();
    expect(getByText(BOOKS_MOCK[0].genre)).toBeTruthy();
  });
  it('Render with all properties and image uri', () => {
    const { getByLabelText, getByText } = render(<Detail book={BOOKS_MOCK[2]} />);
    const image = getByLabelText('detailImage');
    expect(image.props.source).toEqual({ uri: BOOKS_MOCK[2].imageUrl });
    expect(getByText(BOOKS_MOCK[2].title)).toBeTruthy();
    expect(getByText(BOOKS_MOCK[2].author)).toBeTruthy();
    expect(getByText(BOOKS_MOCK[2].year)).toBeTruthy();
    expect(getByText(BOOKS_MOCK[2].genre)).toBeTruthy();
  });
});
