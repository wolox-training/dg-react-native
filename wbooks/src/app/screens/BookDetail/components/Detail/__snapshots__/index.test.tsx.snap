// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail test suit Render correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#EAF6FA",
      "flex": 1,
      "paddingVertical": 10,
    }
  }
>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "flex": 1,
            "padding": 20,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 5,
              "elevation": 1,
              "paddingHorizontal": 30,
              "paddingVertical": 15,
              "shadowColor": "#030303",
              "shadowOffset": Object {
                "height": 1,
                "width": 0,
              },
              "shadowOpacity": 0.18,
              "shadowRadius": 1,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
                "marginBottom": 30,
              }
            }
          >
            <Image
              accessibilityLabel="detailImage"
              resizeMode="stretch"
              source={
                Object {
                  "process": [Function],
                }
              }
              style={
                Object {
                  "height": 105,
                  "width": 69,
                }
              }
            />
            <View>
              <Text
                numberOfLines={1}
                style={
                  Object {
                    "color": "#4A4A4A",
                    "fontSize": 24,
                    "marginBottom": 5,
                    "paddingLeft": 20,
                    "paddingRight": 50,
                  }
                }
              >
                Ring of Bright Water: The revenge of the dark fire ring
              </Text>
              <Text
                style={
                  Object {
                    "color": "#4A4A4A",
                    "fontSize": 15,
                    "paddingLeft": 20,
                  }
                }
              >
                Emmie Thiel
              </Text>
              <Text
                style={
                  Object {
                    "color": "#4A4A4A",
                    "fontSize": 15,
                    "paddingLeft": 20,
                  }
                }
              >
                1968
              </Text>
              <Text
                style={
                  Object {
                    "color": "#4A4A4A",
                    "fontSize": 15,
                    "paddingLeft": 20,
                  }
                }
              >
                Short story
              </Text>
            </View>
          </View>
          <View
            accessibilityLabel="btn"
            accessible={true}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#FFFFFF",
                "borderColor": "#00ADEE",
                "borderRadius": 22,
                "borderWidth": 1,
                "height": 44,
                "justifyContent": "center",
                "opacity": 1,
                "width": "100%",
              }
            }
          >
            <Text
              accessibilityLabel="btnTitle"
              style={
                Array [
                  Object {
                    "fontSize": 15,
                  },
                  Object {
                    "color": "#00ADEE",
                  },
                ]
              }
            >
              ADD TO WISHLIST
            </Text>
          </View>
          <View
            accessibilityLabel="btn"
            accessible={true}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#00ADEE",
                "borderColor": "#00ADEE",
                "borderRadius": 22,
                "borderWidth": 1,
                "height": 44,
                "justifyContent": "center",
                "marginTop": 10,
                "opacity": 1,
                "width": "100%",
              }
            }
          >
            <Text
              accessibilityLabel="btnTitle"
              style={
                Array [
                  Object {
                    "fontSize": 15,
                  },
                  Object {
                    "color": "#FFFFFF",
                  },
                ]
              }
            >
              RENT
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
